Tests/AmplitudeTests/AmplitudeTests.swift

```diff
@@ -534,6 +534,82 @@ final class AmplitudeTests: XCTestCase {
         legacyStorageAmplitude.identifyStorage.reset()
     }
     #endif
+    
+    func testRemnantDataNotMigratedInNonSandboxedApps() throws {
+        let instanceName = "legacy_v3_\(UUID().uuidString)".lowercased()
+        let bundle = Bundle(for: type(of: self))
+        let legacyDbUrl = bundle.url(forResource: "legacy_v3", withExtension: "sqlite")
+        let dbUrl = LegacyDatabaseStorage.getDatabasePath(instanceName)
+        let fileManager = FileManager.default
+        let legacyDbExists = legacyDbUrl != nil ? fileManager.fileExists(atPath: legacyDbUrl!.path) : false
+        XCTAssertTrue(legacyDbExists)
+
+        try fileManager.copyItem(at: legacyDbUrl!, to: dbUrl)
+
+        addTeardownBlock {
+            let fileManager = FileManager.default
+            if fileManager.fileExists(atPath: dbUrl.path) {
+                try fileManager.removeItem(at: dbUrl)
+            }
+        }
+        
+        let apiKey = "test-api-key"
+        let configuration = Configuration(
+            apiKey: apiKey,
+            instanceName: instanceName,
+            migrateLegacyData: true
+        )
+        let amplitude = Amplitude(configuration: configuration)
+
+        let deviceId = "9B574574-74A7-4EDF-969D-164CB151B6C3"
+        let userId = "ios-sample-user-legacy"
+               
+        #if os(macOS)
+            // We don't want to transfer remnant data in non-sanboxed apps
+            XCTAssertFalse(amplitude.isSandboxEnabled())
+            XCTAssertNotEqual(amplitude.getDeviceId(), deviceId)
```

Comment:
In this case should we still transfer the user/deviceId info (userdefaults) but not transfer events?

That is what I did here - https://github.com/amplitude/Amplitude-Swift/pull/114/files#diff-e6394ef8945b50a2df6dcadfe94d9f07aac5004023649149509051a286c740daR14-R23

Reply:
Nevermind, I forgot that maintenance iOS stores id in the same DB by instance name. We should skip both id and events.